### PR TITLE
Feature/scan tool gui

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Creates a GUI for the scan tool which shows the position, resources and enables the user to choose a name
- The positions can be changed only in gamemode 1 (to avoid scanning submerged areas with it)

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/items/ItemScanTool.java [hunk 1]
```diff
@@ -108,8 +109,20 @@ else if (!compound.hasKey(SECOND_POS_STRING))
             //todo if on client (ssp) -> no need to send message, we can execute it in worldIn.isRemote without a message.
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/items/ItemScanTool.java:109]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowScan.java [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+                    final BlockPos here = new BlockPos(x, y, z);
+                    final IBlockState blockState = world.getBlockState(here);
+                    final TileEntity tileEntity = world.getTileEntity(here);
+                    final List<Entity> list = world.getEntitiesWithinAABB(Entity.class, new AxisAlignedBB(here));
+
+                    for (final Entity entity : list)
+                    {
+                        for (final ItemStack stack : ItemStackUtils.getListOfStackForEntity(entity, Minecraft.getMinecraft().player))
+                        {
+                            if (!ItemStackUtils.isEmpty(stack))
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowScan.java:213]: CRITICAL Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowScan.java [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+                    }
+
+                    @Nullable final Block block = blockState.getBlock();
+                    if (block != null && block != Blocks.AIR)
+                    {
+                        if (tileEntity != null)
+                        {
+                            final List<ItemStack> itemList = new ArrayList<>();
+                            itemList.addAll(ItemStackUtils.getItemStacksOfTileEntity(tileEntity.writeToNBT(new NBTTagCompound()), world));
+                            for (final ItemStack stack : itemList)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowScan.java:237]: CRITICAL Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowScan.java [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+        {
+            updateResources();
+        }
+        return super.onKeyTyped(ch, key);
+    }
+
+    /**
+     * Clears and resets/updates all resources.
+     */
+    private void updateResources()
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowScan.java]: MAJOR Refactor this method to reduce its Cognitive Complexity from 47 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 63 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 57 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CommonProxy.java#L131](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java#L131): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowBuildBuilding.java#L182](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java#L182): The Cyclomatic Complexity of this method "updateResources" is 26 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [WindowBuildBuilding.java#L182](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java#L182): Refactor this method to reduce its Cognitive Complexity from 43 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WindowBuildBuilding.java#L219](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java#L219): Reduce the total number of break and continue statements in this loop to use at most one. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)
1. ![MAJOR][MAJOR] [WindowBuildBuilding.java#L336](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java#L336): Remove this unused method parameter "list". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/74c39f5b747860450d4904647794b3a11e91526d/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**